torch/csrc/jit/passes/memory_planning/memory_observer.cpp

```diff
@@ -0,0 +1,165 @@
+//
+// Created by Maksim Levental on 9/11/21.
+//
+
+#include <iostream>
+#include <utility>
+
+#include <torch/csrc/jit/ir/ir.h>
+#include <torch/csrc/jit/passes/memory_planning/memory_observer.h>
+#include <torch/csrc/jit/runtime/interpreter.h>
+
+namespace torch {
+namespace jit {
+
+MemoryObserverThreadLocalState* getMemoryObserverTLSState() {
+  const auto& state =
+      c10::ThreadLocalDebugInfo::get(c10::DebugInfoKind::PROFILER_STATE);
+  return static_cast<MemoryObserverThreadLocalState*>(state);
+}
+
+struct MemoryObserverContext : public at::ObserverContext {
```

Comment:
we should consider roping in people with a little more context here to the at::ObserverContext... i certainly do not have it